packages/injective-cosmwasm/src/exchange_mock_querier.rs

```diff
@@ -648,6 +648,37 @@ pub mod handlers {
         Some(Box::new(Temp { orders, assertion }))
     }
 
+    pub type DerivativeUpToAmountConsumingFunction = fn(MarketId, SubaccountId, FPDecimal, i32, Option<FPDecimal>);
+
+    pub fn create_derivative_orders_up_to_amount_handler(
+        orders: Option<Vec<TrimmedDerivativeLimitOrder>>,
+        assertion: Option<DerivativeUpToAmountConsumingFunction>,
+    ) -> Option<Box<dyn HandlesTraderDerivativeOrdersToCancelUpToAmountQuery>> {
+        struct Temp {
+            orders: Option<Vec<TrimmedDerivativeLimitOrder>>,
+            assertion: Option<DerivativeUpToAmountConsumingFunction>,
+        }
+        impl HandlesTraderDerivativeOrdersToCancelUpToAmountQuery for Temp {
+            fn handle(
+                &self,
+                market_id: MarketId,
+                subaccount_id: SubaccountId,
+                quote_amount: FPDecimal,
+                strategy: i32,
+                reference_price: Option<FPDecimal>,
+            ) -> QuerierResult {
+                if self.assertion.is_some() {
```

Comment:
This assertion is to panic query in some specific conditions?  do we rely on such behavior in case of real query that we want to simulate it in the handler?

Reply:
Just an additional optional assertion for unit-tests to check that the query was called with the expected data.